pkg/certificate/manager_test.go

```diff
@@ -157,6 +182,32 @@ var _ = Describe("certificate manager", func() {
 		ExpectWithOffset(1, err).To(Succeed(), "should success updating mutatingwebhookconfiguration")
 	}
 
+	DescribeTable("VerifyLabels",
+		func(c verifyLabelsTestCase) {
+			createResources()
+			defer deleteResources()
+			manager := c.createManager()
+			obtainedCASecret := loadCASecret(manager)
+			obtainedSecret := loadServiceSecret(manager)
+
+			if c.shouldHaveLabels {
+				Expect(obtainedCASecret.GetLabels()[expectedLabelKey]).To(Equal(expectedLabelValue))
+				Expect(obtainedSecret.GetLabels()[expectedLabelKey]).To(Equal(expectedLabelValue))
+			} else {
+				Expect(obtainedCASecret.GetLabels()).To(HaveLen(0))
+				Expect(obtainedSecret.GetLabels()).To(HaveLen(0))
+			}
+		},
+		Entry("when manager created with labels, should have labels", verifyLabelsTestCase{
+			createManager:    newManagerWithLabels,
+			shouldHaveLabels: true,
+		}),
+		Entry("when manager created without labels, should have no labels", verifyLabelsTestCase{
+			createManager:    newManager,
+			shouldHaveLabels: false,
+		}),
+	)
+
```

Comment:
There is a pair of tests here it does not worth it a DescriptionTable half of the function is different per test, please create individual Specs

```golang
Expect(obtainedCASecret.GetLabels()).To(HaveKeyWithValue(expectedLabelKey, expectedLabelValue)
```

and

```golang
Expect(obtainedCASecret.GetLabels()).To(BeEmpty())
```